Add GitHub-related configuration

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request adds some core GitHub configuration files from [cisagov/skeleton-docker](https://github.com/cisagov/skeleton-docker). This includes:

- dependabot configuration file
- CodeQL workflow
- CODEOWNERS file
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

This sets up some bare bones security maintenance and will alert our team if anyone were to create a pull request in this repository.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

We don't need no testin'.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).

mcdonnnj committed:
Add some core GitHub maintenance files to this repository

This includes a CODEOWNERS file, dependabot configuration, and CodeQL
workflow.

jsf9k reviewed (approved):
Ad astra!

dav3r commented on .github/CODEOWNERS:
```diff
@@ -0,0 +1,10 @@
+# These owners will be the default owners for everything in the
+# repo. Unless a later match takes precedence, these owners will be
+# requested for review when someone opens a pull request.
+* @dav3r @felddy @jsf9k @mcdonnnj
```
I don't believe this code currently has any owners in the cisagov org.  Definitely agree that our team should own the `.github/` directory.
```suggestion
```

dav3r reviewed (changes requested):
Please see my small, but important note.

mcdonnnj committed:
Refine CODEOWNERS configuration

The only thing we're willing to own is the `.github/` directory.

Co-authored-by: dav3r <daver@geekpad.com>

mcdonnnj committed:
Use updated CodeQL workflow

This pulls in an updated CodeQL workflow from
cisagov/skeleton-python-library.

dav3r reviewed (approved):
👍 👍 🚀

mcdonnnj committed:
Add `javascript` as a language checked by CodeQL

There is some JavaScript in this repository so it should be checked.

mcdonnnj committed:
Update dependabot configuration

This modifies the docker and pip configuration to reflect the
organization of this repository. It also adds checking for npm since
this project contains a package.json file.

jsf9k reviewed (approved):
Still LGTM
